internal/users/user_cache.go

```diff
@@ -0,0 +1,42 @@
+package users
+
+import (
+	"sync"
+)
+
+type UserCache struct {
+	data map[string]User
+	mu   sync.RWMutex
+}
+
+var instance *UserCache
+var once sync.Once
+
+func GetUserCache() *UserCache {
```

Comment:
Calling this UserCache is confusing as you would expect this to be for cached object, not necessarily an actual user store. 

An interface called UserStore and then an implementation called InMemoryUserStore would make more sense to me.